tests/topotato/bgp-test_community_alias.py

```diff
@@ -0,0 +1,376 @@
+        exit-address-family
+        !
+        ip prefix-list p1 permit {{ routers.r2.lo_ip4[0] }}
+        ip prefix-list p2 permit {{ routers.r2.lo_ip4[1] }}
+        ip prefix-list p3 permit {{ routers.r2.lo_ip4[2] }}
+        !
+        route-map r1 permit 10
+            match ip address prefix-list p1
+            set community 65001:1 65001:2
+            set large-community 65001:1:1 65001:1:2
+        route-map r1 permit 20
+            match ip address prefix-list p2
+            set community 65002:1 65002:2
+        route-map r1 permit 30
+            match ip address prefix-list p3
+        !
+        #%   endif
+    #% endblock
+    """
+
+
+@config_fixture(Configs)
+def configs(config, allproto_topo):
+    return
+
+
+@instance_fixture()
+def testenv(configs):
+    return FRRNetworkInstance(configs.topology, configs).prepare()
+
+
+class BGPCommunityAliasTest(TestBase):
+    instancefn = testenv
+
+    @topotatofunc
+    def _bgp_converge(self, topo, r1, r2):
+        expected = {
+            str(r2.lo_ip4[0]): [
+                {
+                    "tag": 10,
+                    "communities": "community-r2-1 65001:2",
+                    "largeCommunities": "large-community-r2-1 65001:1:2",
+                }
+            ],
+            str(r2.lo_ip4[1]): [
+                {
+                    "tag": 20,
+                    "communities": "65002:1 community-r2-2",
+                    "largeCommunities": "",
+                }
+            ],
+            str(r2.lo_ip4[2]): [
+                {
+                    "tag": 100,
+                    "communities": "",
+                    "largeCommunities": "",
+                }
+            ],
+        }
+
+        yield from AssertVtysh.make(
+            r1,
+            "zebra",
+            f"show ip route json",
+            maxwait=5.0,
+            compare=expected,
+        )
+
+    @topotatofunc
+    def _bgp_show_prefixes_by_alias(self, topo, r1, r2):
+        expected = {
+            "routes": {
+                str(r2.lo_ip4[0]): [
+                    {
+                        "community": {"string": "community-r2-1 65001:2"},
+                        "largeCommunity": {"string": "large-community-r2-1 65001:1:2"},
+                    }
+                ]
+            }
+        }
+
+        yield from AssertVtysh.make(
+            r1,
+            "bgpd",
+            f"show bgp ipv4 unicast alias large-community-r2-1 json detail",
+            maxwait=5.0,
+            compare=expected,
+        )
+
+    @topotatofunc
+    def _bgp_show_prefixes_by_large_community_list(self, topo, r1, r2):
+        expected = {"routes": {str(r2.lo_ip4[0]): [{"valid": True}]}}
+
+        yield from AssertVtysh.make(
+            r1,
+            "bgpd",
+            f"show bgp ipv4 unicast large-community-list r2 json",# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2022 Tsen Chee Vincent LEUNG YIN KO
```

Comment:
Did something go wrong in rebasing here? The file seems to be duplicated :laughing:

Reply:
Good catch, my bad